src/components/pages/Contacts/index.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+import { RootState } from 'src/redux';
+
+import { bindActionCreators, Dispatch } from 'redux';
+import { connect, ConnectedProps } from 'react-redux';
+import {
+  loadOrgContacts,
+  addFilter,
+  removeFilter,
+} from 'src/redux/modules/orgcontacts';
+import { loadTags } from 'src/redux/modules/tag';
+
+import TagSelector from 'src/components/tags/TagSelector';
+import FormControl from 'react-bootstrap/FormControl';
+import Form from 'react-bootstrap/Form';
+import Table from 'react-bootstrap/Table';
+import Tag from 'src/components/tags/Tag';
+import './index.css';
+import { Link } from 'react-router-dom';
+
+const mapStateToProps = (state: RootState) => ({
+  tags: state.tags.tags,
+  orgContacts: state.orgContacts.contacts,
+  filters: state.orgContacts.selectedFilters,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) =>
+  bindActionCreators(
+    { loadOrgContacts, addFilter, removeFilter, loadTags },
+    dispatch,
+  );
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+const UnconnectedContacts: React.FC<PropsFromRedux> = ({
+  loadTags,
+  tags,
+  orgContacts,
+  filters,
+  loadOrgContacts,
+  addFilter,
+  removeFilter,
+}) => {
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [filteredOrgContact, setFilteredOrgContacts] = useState<OrgContact[]>(
+    [],
+  );
+  const [hasFetchedData, setHasFetchedData] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!hasFetchedData && orgContacts.length === 0) {
+      loadOrgContacts();
+      loadTags();
```

Comment:
do you want to `await` either of these calls?

Reply:
synced offline! looks like await is not needed here